packages/ui5-middleware-serveframework/lib/index.js

```diff
@@ -73,16 +74,24 @@ module.exports = async ({ log, options, middlewareUtil }) => {
 
 			// support for coporate proxies
 			const { getProxyForUrl } = await import("proxy-from-env");
-			const proxyUrl = getProxyForUrl(baseUrl);
 			const { HttpsProxyAgent } = await import("https-proxy-agent");
-			const agent = proxyUrl ? new HttpsProxyAgent(proxyUrl) : undefined;
-			effectiveOptions.debug && log.info(`[${baseUrl}] Proxy: ${proxyUrl ? proxyUrl : "n/a"}`);
+			const https = await import("https");
+
+			const proxyUrl = getProxyForUrl(baseUrl);
+			const agentOptions = { rejectUnauthorized: effectiveOptions.strictSSL };
+
+			const agent = proxyUrl ? new HttpsProxyAgent(new URL(proxyUrl), agentOptions) : new https.Agent(agentOptions);
+
+			if (effectiveOptions.debug) {
+				log.info(`[${baseUrl}] Proxy: ${proxyUrl || "n/a"}, strictSSL: ${effectiveOptions.strictSSL}`);
+			}
 
 			// fetch the version information for the concrete version from CDN
 			const fetch = (await import("node-fetch")).default;
-			await fetch(`${baseUrl}/${frameworkVersion}/resources/sap-ui-version.json`, { agent })
-				.then((res) => res.json())
-				.then((json) => writeFile(versionInfoFile, JSON.stringify(json), { encoding: "utf-8" }));
+			const versionUrl = `${baseUrl}/${frameworkVersion}/resources/sap-ui-version.json`;
+			const response = await fetch(versionUrl, { agent });
```

Comment:
Now I understand it - here you pass the native HttpsAgent with the option `rejectUnauthorized` and this disables the ssl verification for fetch

👍